ts/Extensions/MouseWheelZoom/MouseWheelZoom.ts

```diff
@@ -173,91 +213,96 @@ const zoomBy = function (
-                x: newXMin,
-                y: newYMin,
-                width: newXRange,
-                height: newYRange
-            }, {
-                x: outerX,
-                y: outerY,
-                width: outerWidth,
-                height: outerHeight
-            }),
+            newExt = fitToRange(
+                outerY,
+                outerHeight,
+                newYMin,
+                newYRange
+            ),
             zoomOut = (
-                newExt.x <= outerX &&
-                newExt.width >=
-                outerWidth &&
-                newExt.y <= outerY &&
-                newExt.height >= outerHeight
+                newExt.rangeStart <= pick(yAxis.options.min, outerY) &&
+                newExt.rangeWidth >= outerHeight &&
+                newExt.rangeStart + newExt.rangeWidth <=
+                    pick(yAxis.options.max, Number.MIN_VALUE)
             );
 
-        // Zoom
-        if (defined(howMuch) && !zoomOut) {
-            if (zoomX) {
-                xAxis.setExtremes(newExt.x, newExt.x + newExt.width, false);
-                hasZoomed = true;
-            }
-            if (zoomY) {
-                yAxis.setExtremes(newExt.y, newExt.y + newExt.height, false);
+        if (zoomY) {
+            if (defined(howMuch) && !zoomOut) { // Zoom
+                yAxis.setExtremes(
+                    newExt.rangeStart,
+                    newExt.rangeStart + newExt.rangeWidth,
+                    false
+                );
                 hasZoomed = true;
-            }
-
-        // Reset zoom
-        } else {
-            if (zoomX) {
-                xAxis.setExtremes(void 0, void 0, false);
-            }
-            if (zoomY) {
+            } else { // Reset zoom
                 yAxis.setExtremes(void 0, void 0, false);
             }
         }
+    }
 
-        if (hasZoomed) {
-            chart.redraw(false);
-        }
+    return hasZoomed;
+};
+
+
+/**
+ * @private
+ */
+const zoomBy = function (
+    chart: Chart,
+    howMuch: number,
+    centerXArg: number,
+    centerYArg: number,
+    mouseX: number,
+    mouseY: number,
+    options: MouseWheelZoomOptions
+): boolean {
+    const xAxis = chart.xAxis[0],
+        yAxis = chart.yAxis[0],
+        type = pick(
+            options.type,
+            chart.zooming.type,
+            ''
+        ),
+        zoomX = /x/.test(type),
+        zoomY = /y/.test(type);
+
+
+    if (chart.inverted) {
+        const emulateRoof = yAxis.pos + yAxis.len;
+
+        // Get the correct values
+        centerXArg = xAxis.toValue(mouseY);
+        centerYArg = yAxis.toValue(mouseX);
+
+        // Swapping x and y for simplicity when chart is inverted.
+        const tmp = mouseX;
+        mouseX = mouseY;
+        mouseY = emulateRoof - tmp + yAxis.pos;
+    }
+
+    const hasZoomedX = zoomOnX(chart, zoomX, mouseX, howMuch, centerXArg),
+        hasZoomedY = zoomOnY(chart, zoomY, mouseY, howMuch, centerYArg),
```

Comment:
If you change the `zoomOnX` and `zoomOnY` to be the same method,  you can use them like that
```suggestion
    const hasZoomedX = zoomOnDir(chart, AzoomX, mouseX, howMuch, centerXArg),
        hasZoomedY = zoomOnDir(chart, zoomY, mouseY, howMuch, centerYArg),
```

you probably need to pass the correct axis as well 🤔🤨